ed_sensor_integration/tools/fitter_viz.cpp

```diff
@@ -0,0 +1,218 @@
+
+}
+
+cv::Mat visualizeFitting(EntityRepresentation2D entity, geo::Transform2 sensor_pose, geo::Transform2 entity_pose,
+                         geo::Transform2 fitted_pose, FitterData fitterdata, bool estimateEntityPose)
+{
+    // visualize fitting
+    cv::Mat canvas = cv::Mat(canvas_height, canvas_width, CV_8UC3, cv::Scalar(0, 0, 0));
+
+    // paint to screen the location of the sensor
+    cv::circle(canvas, sensorlocation, 3, sensor_colour, cv::FILLED);
+
+    // paint entity (from worldmodel)
+    geo::Transform2 relpose = sensor_pose.inverse() * entity_pose;
+    drawShape2D(canvas, entity.shape_2d, relpose, canvas_resolution, sensor_x, sensor_y, entity_colour);
+
+    if (estimateEntityPose)
+    {
+        // paint fitted entity
+        geo::Transform2 fitted_relpose = sensor_pose.inverse() * fitted_pose;
+        drawShape2D(canvas, entity.shape_2d, fitted_relpose, canvas_resolution, sensor_x, sensor_y, fitted_colour);
+    }
+    else
+    {
+        std::cout << "Fitted entity not printed in image because of a fitter error" << std::endl;
+    }
+
+    // paint sensor_ranges
+    uint nranges = fitterdata.sensor_ranges.size();
+    uint half_nranges = nranges/2;
+    float fx = fitterdata.fx;
+    for (unsigned int i = 0; i < nranges; ++i)
+    {
+        float x_m = fitterdata.sensor_ranges[i] * ((static_cast< float >(i) - half_nranges) / fx);
+        float y_m = fitterdata.sensor_ranges[i];
+
+        // postion to pixels
+        int x_p = sensor_x + (int)(x_m * canvas_resolution);
+        int y_p = sensor_y - (int)(y_m * canvas_resolution);
+
+        if (x_p < 0 || x_p >= canvas_width)
+            continue;
+        if (y_p < 0 || y_p >= canvas_height)
+            continue;
+        if (fitterdata.sensor_ranges[i] == 0) // filter out sensor_ranges equal to zero
+            continue;
+
+        // paint to screen
+        cv::Point centerCircle(x_p, y_p);
+        cv::circle(canvas, centerCircle, 2, measurement_colour, cv::FILLED);
+    }
+
+    return canvas;
+}
+
+int main(int argc, char **argv)
+{
+    if (argc != 4)
+    {
+        usage();
+        return 1;
+    }
+
+    std::string model_name = argv[2];
+
+    ed::WorldModel world_model;
+    if (!ed_sensor_integration::loadWorldModel(model_name, world_model)){
+        std::cerr << "World model '" << model_name << "' could not be loaded." << std::endl;
+        return 1;
+    }
+
+    std::string entity_id = argv[3];
+    ed::EntityConstPtr e = world_model.getEntity(entity_id);
+    if (!e)
+    {
+        std::cerr << "Entity '" << entity_id << "' could not be found in world model '" << model_name << "'." << std::endl;
+        return 1;
+    }
+
+    tue::filesystem::Path path = argv[1];
+    if (!path.exists())
+    {
+        std::cerr << "Path '" << path << "' does not exist." << std::endl;
+        return 1;
+    }
+
+    ed_sensor_integration::SnapshotCrawler crawler(path);
+
+    Fitter fitter;
+
+    while(true)
```

Comment:
Why? I think `ros::ok()` is better.

Reply:
Because this executable does nothing with ros